packages/bpk-component-navigation-bar/src/BpkNavigationBarButtonLink.tsx

```diff
@@ -16,17 +16,15 @@
  * limitations under the License.
  */
 
-import type { ComponentProps, MouseEvent, ReactNode } from 'react';
-
-import { BpkButtonLink } from '../../bpk-component-link';
+import BpkLink, { type BpkLinkProps } from '../../bpk-component-link';
 
 import { BAR_STYLES, type BarStyle } from './BpkNavigationBar';
 
-export interface Props extends ComponentProps<typeof BpkButtonLink> {
-  children: ReactNode,
-  onClick: (event: MouseEvent<HTMLElement>) => void,
-  className?: string,
-  barStyle?: BarStyle,
+export interface Props extends Omit<BpkLinkProps<'button'>, 'as'> {
```

Comment:
Here `Props = BpkLink` as all properties of the button - as property + custom properties (className, barStyle, ect). So `BpkNavigationBarButtonLink` can accept all the button related properties (such as onClick, disabled, aria - label, etc.), while hiding the as attribute (for internal fixation for the 'button').